src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,105 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Transferring between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` transferred to lie in `H` -/
+@[protected, simp] def transfer : Π {u v : V} (p : G.walk u v) (H : simple_graph V)
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons' u v w a p) H h :=
+  walk.cons (h (⟦(u, v)⟧ : sym2 V) (by simp)) (p.transfer H (λ e he, h e (by simp [he])))
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma transfer_id : p.transfer G (λ e ep, edges_subset_edge_set p ep) = p :=
+by { induction p; simp only [*, transfer, eq_self_iff_true, heq_iff_eq, and_self], }
+
+/-- The walk `p` transfered to the larger graph `H` -/
+abbreviation transfer_le (GH : G ≤ H) : H.walk u v :=
+p.transfer H (λ e ep, edge_set_mono GH (edges_subset_edge_set p ep))
+
+lemma transfer_eq_map_spanning_subgraphs (GH : G ≤ H) :
+  p.transfer H hp = p.map (simple_graph.hom.map_spanning_subgraphs GH) :=
+by { induction p; simp only [*, transfer, map_cons, hom.map_spanning_subgraphs_apply,
+                             eq_self_iff_true, heq_iff_eq, and_self, map_nil], }
+
+@[simp] lemma edges_transfer : (p.transfer H hp).edges = p.edges :=
+by { induction p; simp only [*, transfer, edges_nil, edges_cons, eq_self_iff_true, and_self], }
+
+@[simp] lemma support_transfer : (p.transfer H hp).support = p.support :=
+by { induction p; simp only [*, transfer, eq_self_iff_true, and_self, support_nil, support_cons], }
+
+lemma is_path_transfer (pp : p.is_path) : (p.transfer H hp).is_path :=
+begin
+  induction p;
+  simp only [transfer, is_path.nil, cons_is_path_iff, support_transfer] at pp ⊢,
+  { tauto, },
+end
+
+lemma is_cycle_transfer (p : G.walk u u) (hp) (pc : p.is_cycle) : (p.transfer H hp).is_cycle :=
+begin
+  cases p;
+  simp only [transfer, is_cycle.not_of_nil, cons_is_cycle_iff, transfer, edges_transfer] at pc ⊢,
+  { exact pc, },
+  { refine ⟨is_path_transfer _ _ pc.left, pc.right⟩, },
+end
+
+lemma is_cycle_transfer_le (p : G.walk u u) (hp) (pc : p.is_cycle)
+  (GH : G ≤ H) : (p.transfer H hp).is_cycle :=
+p.is_cycle_transfer (λ e ep, edge_set_mono GH (edges_subset_edge_set p ep)) pc
```

Comment:
I haven't gotten rid of `induce_le` yet, essentially because it seems less "overkill" than `p.map (simple_graph.hom.map_spanning_subgraphs GH)` (and it's much shorter to type!), but also: if I'm to get rid of it, where would you put this `is_cycle_transfer_le`, restated in terms of `map_spanning_subgraphs GH`.
If keeping `induce_le` is negotiable, I'd keep it, otherwise would welcome a proposal for where to fit this lemma (I actually use it in the tree stuff, hence it appearing here -- btw, I should probably also add the `is_path` version). 
Would just leaving it here work, even if it (superficially at least) is not related to `induce` ?

Reply:
Rather than `induce_le`, I'd like to have
```
/-- The specialization of `simple_graph.walk.map` for mapping walks to supergraphs. -/
@[reducible] def map_le {G G' : simple_graph V} (h : G ≤ G')
  {u v : V} (p : G.walk u v): G'.walk u v :=
p.map (hom.map_spanning_subgraphs h)
```
which can go right after the definition of `map`. I think we should prefer `map` whenever possible.

By the way, a more general statement of this cycle transfer lemma would be an analogue of `map_is_path_of_injective` for cycles. (And also that means you don't need to add an `is_path` version.)

Reply:
OK, then I'll just remove `induce_le`, and I think `map_le` and `map_is_path_of_injective` probably belong to another PR?

Reply:
It probably would have been fine to slip in `map_le` into this PR, but creating a followup PR is fine too, so now that you've removed it let's do that.

Reply:
OK, working on it

Reply:
Bah, probably not done with it this week-end. I figured the easiest way for `map_is_cycle_of_injective` and friends would be to have a lemma `p.is_cycle <-> p.length ≥ 3 /\ p.support.nodup` but then I also figured the lemma `is_path_def : p.is_path <-> p.support.nodup` should **also** be used to prove `map_is_path_of_injective` and friends, because it would simplify the proofs, but I think for that to work I'd need `l.nodup <-> l.length = l.to_finset.card`, which seemingly isn't there?

Reply:
They still need refinement (and might have unnecessary steps!), but here's what I came up with:
```lean
lemma map_is_trail_of_injective (hinj : function.injective f) (hp : p.is_trail) :
  (p.map f).is_trail :=
begin
  rw is_trail_def at hp ⊢,
  induction p with w u v w huv hvw ih,
  { simp, },
  { simp at hp,
    specialize ih hp.2,
    rw [map_cons, edges_cons, list.nodup_cons],
    split,
    { rw [edges_map],
      change sym2.map f ⟦(u, v)⟧ ∉ _,
      rw list.mem_map_of_injective (sym2.map.injective hinj),
      exact hp.1, },
    { exact ih, }, },
end

@[simp] lemma map_eq_nil_iff {p : G.walk u u} : p.map f = nil ↔ p = nil :=
begin
  split,
  { intro h,
    cases p,
    { refl },
    { injection h } },
  { rintro rfl,
    refl, }
end

lemma map_is_cycle_of_injective {p : G.walk u u} (hinj : function.injective f) (hp : p.is_cycle) :
  (p.map f).is_cycle :=
begin
  rw is_cycle_def at hp ⊢,
  simp [map_is_trail_of_injective hinj hp.1],
  split,
  { intro h,
    apply hp.2.1 h, },
  { replace hp := hp.2.2,
    rwa [← list.map_tail, list.nodup_map_iff hinj], }
end
```